source/common/upstream/outlier_detection_impl.cc

```diff
@@ -209,8 +219,51 @@ void DetectorImpl::onConsecutive5xxWorker(HostPtr host) {
 
 void DetectorImpl::onIntervalTimer() {
```

Comment:
maybe it's a good time to start thinking about small test setup where you can feed upstream hosts params and run ejection on that group of hosts, similar to the idea (https://github.com/lyft/envoy/blob/master/test/common/upstream/load_balancer_simulation_test.cc#L22) of what we are doing to validate how AZ routing works.

With more complicated outlier detection algos we have to have a way to run simulation.